src/protobuf-net.Reflection/Internal/CodeGen/CodeGenEnum.cs

```diff
@@ -14,8 +15,8 @@ public CodeGenEnum(string name, string fullyQualifiedPrefix) : base(name, fullyQ
 
     public string OriginalName { get; set; }
     
-    private List<CodeGenEnumValue>? _enumValues;
-    public List<CodeGenEnumValue> EnumValues => _enumValues ??= new();
+    private NonNullableList<CodeGenEnumValue>? _enumValues;
```

Comment:
it *seems* like the NRT checking should have found any problems we had here; is is possible that we didn't have NRT enabled in the code that inspired this change?

to be clear: I'm not massively opposed - it just seems a little overkill

Reply:
Probably, we could find some troubles. But we can still inject this error-output in side of `NonNullableList<T>.Add()`, so you will get the error output \ exception \ anything you want there.

From another point of view - this helps to write less code and dont make an error during the `CodeGenXXX` models parsing. For example code below will have to be modified when addition of item happens, and I guess the probability of forgetting to handle `null` here is very high.
```
if (childSymbol is ITypeSymbol typeSymbol)
            {
                switch (typeSymbol.TypeKind)
                {
                    case TypeKind.Struct:
                    case TypeKind.Class:
                        var messageParser = ParserProvider.GetMessageParser(_namespaceParseContext);
                        var nestedMessage = messageParser.Parse(typeSymbol);

// MAKE SURE THIS IS NOT NULL
                        codeGenMessage.Messages.Add(nestedMessage);
                        break;
                    
                    case TypeKind.Enum:
                        var enumParser = ParserProvider.GetEnumParser();
                        var nestedEnum = enumParser.Parse(typeSymbol);

// MAKE SURE THIS IS NOT NULL
                        codeGenMessage.Enums.Add(nestedEnum);
                        break;
                }
            }
```
Personally, I am not a fan of reading code with billions of if's, that's why I introduced such a change.
What do you think? If you say you would rewrite - I can rollback this change and handle more cases explicitly